inc/classes/optimization/JS/class-combine.php

```diff
@@ -112,52 +119,57 @@ public function optimize() {
 	 * @since 3.1
 	 * @author Remy Perona
 	 *
-	 * @param HtmlPageCrawler $nodes NodeS corresponding to JS file or content.
+	 * @param Array $scripts scripts corresponding to JS file or content.
 	 * @return array
 	 */
-	protected function parse( $nodes ) {
-		$combine_nodes = $nodes->each( function( \Wa72\HtmlPageDom\HtmlPageCrawler $node, $i ) {
-			$src = $node->attr( 'src' );
-
-			if ( $src ) {
-				if ( $this->is_external_file( $src ) ) {
-					foreach ( $this->get_excluded_external_file_path() as $excluded_external_file ) {
-						if ( false !== strpos( $src, $excluded_external_file ) ) {
+	protected function parse( $scripts ) {
+		$scripts = array_map( function( $script ) {
+			preg_match( '/<script\s+([^>]+[\s\'\"])?src\s*=\s*[\'\"]\s*?([^\'\"]+\.js(?:\?[^\'\"]*)?)\s*?[\'\"]([^>]+)?\/?>/Umsi', $script[0], $matches );
```

Comment:
No need to escape double quotes: `[\s\'\"]`, `[\'\"]`, `[^\'\"]` => `[\s"\']`, `["\']`, `[^"\']`.